docs/reference/index-modules.asciidoc

```diff
@@ -82,6 +82,37 @@ breaking change].
     after segments are merged. Segment merging can be forced using
     <<indices-forcemerge,force merge>>.
 
+[[index-mode-setting]] `index.mode`::
++
+The `index.mode` setting is used to control settings applied in specific domains like ingestion of time series data or logs.
+Different mutually exclusive modes exist, which are used to apply settings or default values controlling indexing of documents,
+sorting and other parameters whose value affects indexing or query performance.
++
+[source,console]
+----------------
+PUT my-index-000001
+{
+  "settings": {
+    "index":{
+      "mode":"standard" <1>
+    }
+  }
+}
+----------------
++
+<1> This index uses the `standard` index mode
++
+Index mode supports the following values:
+
+`null`::: Default value (same as `standard`).
+
+`standard`::: Standard indexing with default settings.
+
+`time_series`::: Index mode optimized for storage of metrics documented in <<tsds-index-settings,TSDS Settings>>.
+
+`logs`::: Index mode optimized for storage of logs. It applies default sort settings on the `hostname` and `timestamp` fields and uses <<synthetic-source,synthetic `_source`>>. <<index-modules-index-sorting,Index sorting>> on different fields is still allowed.
```

Comment:
@salvatore-campagna Is this `hostname` or `host.name`? Same for `timestamp` vs `@timestamp`?